File: android/src/main/java/com/bishare/protocol/constants/ServiceTypes.kt
package com.bishare.protocol.constants

/**
 * Bonjour/mDNS service type identifiers for discovery.
 */
object BIShareService {
    /** Main device discovery — with trailing dot for Android NSD */
    const val DISCOVERY = "_bishare._tcp."

    /** Main device discovery — without trailing dot (for matching/comparison) */
    const val DISCOVERY_RAW = "_bishare._tcp"

    /** Transfer room discovery — with trailing dot for Android NSD */
    const val ROOM = "_bishare-room._tcp."

    /** Transfer room discovery — without trailing dot */
    const val ROOM_RAW = "_bishare-room._tcp"

    /** Wi-Fi Aware service type (iOS 26+ WiFiAwareServices naming) */
    const val AWARE = "_bishare-aware._tcp"

    /** Android Wi-Fi Aware (NAN) service name (lowercase, WifiAwareManager) */
    const val AWARE_NAN = "bishare-aware"
}

File: ios/Sources/BIShareProtocol/Constants/ServiceTypes.swift
import Foundation

/// Bonjour/mDNS service type identifiers for discovery.
public enum BIShareService {
    /// Main device discovery service type
    public static let discovery = "_bishare._tcp"

    /// Transfer room discovery service type
    public static let room = "_bishare-room._tcp"

    /// MultipeerConnectivity nearby service type
    public static let nearby = "bishare-nearby"

    /// QUIC ALPN protocol identifier
    public static let quicALPN = ["bishare-quic"]

    /// Wi-Fi Aware service type (iOS 26+ WiFiAwareServices naming)
    public static let aware = "_bishare-aware._tcp"

    /// Android Wi-Fi Aware (NAN) service name, for cross-platform matching
    public static let awareNan = "bishare-aware"
}
